### ISSUE TITLE
batch size not updated with train

### ISSUE THREAD

**Issue by kparichay** [nnstreamer/nntrainer#278]
batch size parameter passed to the train is not reflected in the layers as layers are already initialized when this batchsize it set.

**Comment by jijoongmoon** [comment]
Could you give me more specific about this? weight and gradient can be decided at initialization and hidden buffer is allocated depending on batch size now. So what do you mean we have to reflect the batch size in layer??

**Comment by kparichay** [comment]
Batch size is allowed to be set as a hyperparameter with `ml_train_model_run()`, internally handled by `setTrainConfig`.
At this point, the model is already compiled. With the current implementation, setting batch size at this point has no effect.
So, this has to be removed or updated to support it properly if needed.
